tree/ntuple/v7/src/RColumn.cxx

```diff
@@ -120,6 +133,21 @@ void ROOT::Experimental::Internal::RColumn::MapPage(RClusterIndex clusterIndex)
    // Set fReadPage to an empty page before populating it to prevent double destruction of the previously page in case
    // the page population fails.
    fReadPage = RPage();
-   fReadPage = fPageSource->PopulatePage(fHandleSource, clusterIndex);
+
+   const auto nTeam = fTeam->size();
+   std::size_t iTeam = 1;
+   do {
+      fReadPage = fPageSource->PopulatePage(fTeam->at(fLastGoodTeamIdx)->GetHandleSource(), clusterIndex);
+      if (fReadPage.IsValid())
+         break;
+      fLastGoodTeamIdx = (fLastGoodTeamIdx + iTeam++) % nTeam;
+   } while (iTeam <= nTeam);
+
    R__ASSERT(fReadPage.Contains(clusterIndex));
 }
+
+void ROOT::Experimental::Internal::RColumn::MergeTeams(RColumn &other)
+{
+   fTeam->emplace_back(&other);
+   other.TeamUp(*this);
```

Comment:
This assumes that `other.fTeam` has only one element, `this`, right? Maybe add an `assert` or fully merge the two vectors? Also, I would consider inlining `TeamUp` as just `other.fTeam = fTeam` (unless it is expected to get considerably more complex), right now it's a bit hard to follow the logic of which column is `this` and which one is `other`...